utilities/transactions/pessimistic_transaction_db.cc

```diff
@@ -499,5 +503,183 @@ void PessimisticTransactionDB::UnregisterTransaction(Transaction* txn) {
   transactions_.erase(it);
 }
 
+// Returns true if commit_seq <= snapshot_seq
+bool WritePreparedTxnDB::IsInSnapshot(uint64_t prep_seq,
```

Comment:
Actually it is better if WritePreparedTxnDB don't need to make asumption of external synchronization. Or we need to document the requirement.